crates/issue/src/lib.rs

```diff
@@ -504,36 +505,23 @@ impl<T: Config> Pallet<T> {
         issue: &mut DefaultIssueRequest<T>,
         amount_transferred: Amount<T>,
         expected_total_amount: Amount<T>,
-        maybe_refund_address: Option<BtcAddress>,
-    ) -> Result<Option<H256>, DispatchError> {
+    ) -> Result<(), DispatchError> {
         let surplus_btc = amount_transferred.checked_sub(&expected_total_amount)?;
-
-        match ext::vault_registry::try_increase_to_be_issued_tokens::<T>(&issue.vault, &surplus_btc) {
-            Ok(_) => {
-                // Current vault can handle the surplus; update the issue request
-                Self::set_issue_amount(
-                    &issue_id,
-                    issue,
-                    amount_transferred,
-                    Amount::zero(T::GetGriefingCollateralCurrencyId::get()),
-                )?;
-                Ok(None) // no refund
-            }
-            Err(_) => {
-                // vault does not have enough collateral to accept the over payment, so try refund.
-                if let Some(refund_address) = maybe_refund_address {
-                    ext::refund::request_refund::<T>(
-                        &surplus_btc,
-                        issue.vault.clone(),
-                        issue.requester.clone(),
-                        refund_address,
-                        issue_id.clone(),
-                    )
-                } else {
-                    Ok(None)
-                }
-            }
+        let max_allowed = ext::vault_registry::get_issuable_tokens_from_vault::<T>(&issue.vault)?;
+
+        if let Ok(_) =
+            ext::vault_registry::try_increase_to_be_issued_tokens::<T>(&issue.vault, &surplus_btc.min(&max_allowed)?)
+        {
+            // Current vault can handle the surplus; update the issue request
+            Self::set_issue_amount(
+                &issue_id,
+                issue,
+                amount_transferred,
+                Amount::zero(T::GetGriefingCollateralCurrencyId::get()),
+            )?;
```

Comment:
Sneaky change! It makes sense to do this, but I'd have appreciated a mention in the PR about it.

Anyway, I think you introduced a bug here: if `max_allowed < surplus_btc`, you're setting an issue amount that is too high. Please fix & add a test for this

Reply:
Oh I didn't mean to make any functional changes beyond a rebase but will check.

Reply:
Right I did make this change before, but I'm not sure why I didn't note this in the description. Anyway I've updated the blurb and added some more tests.